Rubberduck.CodeAnalysis/Inspections/ObjectWhereProcedureIsRequiredInspection.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Inspections.Inspections.Extensions;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    /// <summary>
+    /// Identifies places in which an object is used but a procedure is required and a default member exists on the object.
+    /// </summary>
+    /// <why>
+    /// Providing an object in a place in which a procedure is required leads to a call to the objects default member.
+    /// This behavior is not obvious and most likely an error. 
+    /// </why>
+    /// <example hasResult="true">
+    /// <![CDATA[
+    /// Class1:
+    ///
+    /// Public Function Foo() As Long
+    /// Attibute VB_UserMemId = 0
+    ///     Foo = 42
+    /// End Function
+    ///
+    /// Module:
+    /// 
+    /// Public Sub DoSomething(ByVal arg As Class1)
+    ///     arg
```

Comment:
Would `Debug.Print arg` make a better / more realistic example?

Reply:
No, then it is not a procedure coercion.